server/upload.go

```diff
@@ -191,9 +192,13 @@ type ProgressWriter struct {
 	completed int
 	total     int
 	fn        func(api.ProgressResponse)
+	mu        sync.Mutex
```

Comment:
this mutex is necessary otherwise the progress is not accurate. it doesn't affect the download